### ISSUE TITLE
Cannot convert to float

### ISSUE THREAD

**Issue by catzoo** [catzoo/dice.py#1]
Cannot convert the second '(' to a float in this expression:
`(15 - (15 * 0.15)) * 0.08`

The way math.py works is by taking two numbers, converting and solving. Basically, it saw `15 - (` and thought '(' was an number. I thought I had parenthesis working. But I may've forgotten to do inner groups first then the out. Or basically, work on `(15 * 0.15)` rather than the first group.

What it would normally do was solve parenthesis first. For example a simple expression as `15 - (2 + 4)`. It would solve `2+4` first converting it into a number, and then solve the whole expression. 

Basically:
```
15 - (2 + 4)
    2+4 = 6
15 - 6
    15 - 6 = 9

15 - (2+4) = 9
```
